Automattic/wp-calypso
# Gravatar: Implement passwordless signup

Related to 105563-gh-Automattic/gravatar

## Proposed Changes

This PR is to replace the original Gravatar's signup page with the new passwordless one.

* To enable the passwordless signup feature for Gravatar
* To customize the Gravatar's passwordless signup page via the `GravatarStepWrapper` component and the related CSS files
* To clean the related code of the original Gravatar signup page

## Testing Instructions



* Clone this repo, check out this PR, and run it by following [this doc](https://github.com/Automattic/wp-calypso#getting-started)
* Logout your account, login via [this URL](http://calypso.localhost:3000/start/wpcc/oauth2-user?ref=oauth2&oauth2_redirect=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D5777cb757676641a14056297dd37c7fc5c9ede94cbfc02d397e47651e9333b79%26redirect_uri%3Dhttps%253A%252F%252Fen.gravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26jetpack-code%26jetpack-user-id%3D0%26action%3Doauth2-login%26from-calypso%3D1&oauth2_client_id=1854) to test the passwordless flow, it should work as expected:

> Note: You can refer to [this post](https://stackoverflow.com/questions/12750864/temporary-e-mail-accounts-for-integration-tests) to create test email accounts

https://github.com/Automattic/wp-calypso/assets/21308003/f798e59d-d4c0-4fb9-a1f5-240f04579657

* The UI and links should work as expected:

| Desktop | Mobile |
| - | - |
| <img width="794" alt="截圖 2023-08-11 下午5 20 53" src="https://github.com/Automattic/wp-calypso/assets/21308003/93b9c82a-4864-4a97-bbf5-32daf5523262"> | <img width="394" alt="截圖 2023-08-15 下午3 47 04" src="https://github.com/Automattic/wp-calypso/assets/21308003/2aa69287-e27e-4613-8672-2eae3ce946c6"> |

* Now, compare the following items of WPCOM's passwordless signup page between this PR and `trunk`. Both should work the same:
  * Logout your account
  * Switch to the mobile view
  * Check the UI
  * Check the flow, it will be:
    * Signup with your email
    * Activate your email
    * Login with your email
    * Land on your WP.com profile page
  
> Note: You might need to clean the browser's `Cookies` and `IndexedDB` to display the email signup form

## Pre-merge Checklist
